tests/main/validate-container-happy/task.yaml

```diff
@@ -0,0 +1,43 @@
+summary: check the symlinks following the right track
+
+environment:
+    SNAP: test-snapd-validate-container-happy
+    SNAP_INSTALL_DIR: $(pwd)/snap-install-dir
+    SNAP_MOUNT_DIR: $(os.paths snap-mount-dir)
+
+prepare: |
+    snap pack test-snapd-validate-container-happy
+
+execute: |
+
+    # We shouldn't use relative symlinks in Github as they cannot be packed correctly.
+    # So here let's test whether we can still pack such symlinks within a snap and use if needed.
+    # First we "try" to unpack the snap structure and untar the symlinks
+    # Then we pack the snap with these symlinks and then install
+    # Finally we check to see if the symlinks actually support the intervined symlinks
+
+    # Untar the symlinks 
+    mkdir -p "$SNAP_MOUNT_DIR"/"$SNAP"/hell
+    tar -xvf "$SNAP"/hell/hell.tar -C "$SNAP_MOUNT_DIR"/"$SNAP"/hell
+
+    snap try "$SNAP"
+    # Check to see if the symlinks exist
+    readlink "$SNAP_MOUNT_DIR"/"$SNAP"/hell/bar | MATCH "foo -> baz -> qux"
+    readlink "$SNAP_MOUNT_DIR"/"$SNAP"/hell/baz | MATCH "foo -> bar -> qux"
+    readlink "$SNAP_MOUNT_DIR"/"$SNAP"/hell/foo | MATCH "bar -> baz -> qux"
+    readlink "$SNAP_MOUNT_DIR"/"$SNAP"/hell/qux | MATCH "foo -> bar -> baz"
+
+    # Create a new snap structure that includes the unpacked symlinks
+    mkdir -p "$SNAP_INSTALL_DIR"
+    cp -ra "$SNAP"/* "$SNAP_INSTALL_DIR"
+    cp -ra "$SNAP_MOUNT_DIR"/"$SNAP"/hell "$SNAP_INSTALL_DIR"
```

Comment:
I suppose this is fine since you are using `cp -a` and I trust that `cp` is doing the right thing but for clarity, this could section could just be:

```
snap pack "$SNAP"
```